packages/serialization/src/base64.rs

```diff
@@ -0,0 +1,317 @@
+use std::fmt;
+use std::marker::PhantomData;
+
+use serde::{de, ser};
+
+use cosmwasm_std::Binary;
+
+use crate::Serde;
+
+/// Alias of `cosmwasm_std::Binary` for better naming
+pub type Base64 = Binary;
```

Comment:
This renaming does not make much sense to me because an instance of Binary does not hold base64 data. It holds binary data. `Vec::<u8>::from(binary)` is basically a noop, not a base64 decoding. The only place where base64 comes into place is when a `Binary` instance is serialized to JSON.

Reply:
That's fair, but if you think about how it's used in definitions of message types, if i look at these two (oversimplified) type definitions:
```rust
struct Message1 {
    data: Binary,
}

struct Message2 {
    data: Base64,
}
```
I think the second type definition makes it much clearer what format the field should be in the raw JSON message when it's sent as an input or received as output.

I can imagine having an alternative string encoding option `HexStr`, which would behave similarly to `Binary` but de/serialize hexadecimal strings. While from the contract's perspective it would be a transparent wrapper over a `Vec<u8>`, (that it, it's "just binary"), it's distinct in its role in the schema of the contract, and its name should reflect that role. So similarly the `Vec<u8>` wrapper that interprets base64 strings should be called `Base64`. Or `Base64Binary`.